### PR TITLE
Move configuration to setup.cfg and use pytest-cov instead of coverage

### PR DESCRIPTION
**Pull request by henri-hulski**

### CODE REVIEW

File: doc/developing.rst [hunk 1]
```diff
@@ -79,7 +79,8 @@ You can run the tests using `py.test`_::
 
 To generate test coverage information as HTML do::
 
-  $ py.test --cov morepath --cov-report html
```
Comment (faassen) [inline doc/developing.rst]: This change involves turning one line into two. Why not ``py.test --cov --cov-report html``?

File: setup.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
             'flake8',
         ],
         coverage=[
-            'pytest-cov',
+            'coverage',
```
Comment (faassen) [inline setup.py]: Why are we getting rid of pytest-cov? I think pytest-cov has features that exploit pytest, right? Not that we use distributed testing...

Comment (henri-hulski) [inline setup.py]: I think the only reason to use pytest-cov is for distributed testing in which case we would also need xdist. Why use an additional layer, if we don't need it. The coverage syntax is also more declarative and easier to follow. 

For some more arguments see http://stackoverflow.com/a/16524426/5885568. 

I also want a consistent syntax in the docs and for tox. 

And actually at the moment most of our apps use coverage. 

BTW the decision was inspired by more.itsdangerous/ fd47bfc from @href.   ;-)

Comment (henri-hulski) [inline setup.py]: But this is a question of discussion. If we decide to use pytest-cov instead, I would refactor all tox.ini's and change coverage to pytest-cov.

Comment (taschini) [inline setup.py]: Recent versions of ``pytest-cov`` provide a good integration of Coverage into Py.test 
  and I think it makes sense to keep it.

Comment (henri-hulski) [inline setup.py]: >  provide a good integration of Coverage into Py.test

What does that exactly mean?

Comment (henri-hulski) [inline setup.py]: And what are the advantages vs using pure Coverage?

Comment (henri-hulski) [inline setup.py]: @taschini I was researching for some concrete arguments why to use pytest-cov, but found nothing. So if you have some insights please share.

**Comment by henri-hulski** [comment]
Make some tests to compare coverage and pytest-cov, but it seems more or less the same.

Here the tests:

coverage_tox.ini which contais testenvs for coverage, pytest-cov and coverage with pytest-cov dependency::
```ini
[tox]
envlist = coverage,pytest-cov,coverage-pycov
skipsdist = True

[testenv]
basepython = python3.5
usedevelop = True
deps = -efile:{toxinidir}/src/reg#egg=reg
       -efile:{toxinidir}/src/dectate#egg=dectate
       -efile:{toxinidir}/src/importscan#egg=importscan
       -e{toxinidir}[test]
       -r{toxinidir}/requirements/fixtures.txt
commands =
     py.test morepath {posargs}

[testenv:coverage]
basepython = python3.5
deps = {[testenv]deps}
       coverage

commands =
    coverage run --source morepath -m py.test morepath {posargs}
    coverage report -m --fail-under=100

[testenv:pytest-cov]
basepython = python3.5
deps = {[testenv]deps}
       pytest-cov

commands =
    py.test --cov --cov-fail-under=100 {posargs}

[testenv:coverage-pycov]
basepython = python3.5
deps = {[testenv]deps}
       pytest-cov

commands =
    coverage run --source morepath -m py.test morepath {posargs}
    coverage report -m --fail-under=100
```

coverage_perf.sh which times all testenvs twice, the first time it creates the testenv, the second time it uses an already created:
```sh
source env/py3/bin/activate

/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e coverage
/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e pytest-cov
/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e coverage-pycov

/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e coverage
/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e pytest-cov
/usr/bin/time -vao coverage.txt tox -c coverage_tox.ini -e coverage-pycov
```

And here the result with filtered out identical lines (mostly 0)::
```
	Command being timed: "tox -c coverage_tox.ini -e coverage"
	User time (seconds): 42.86
	System time (seconds): 2.94
	Percent of CPU this job got: 79%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:57.28
	Maximum resident set size (kbytes): 102096
	Major (requiring I/O) page faults: 29
	Minor (reclaiming a frame) page faults: 451614
	Voluntary context switches: 1404
	Involuntary context switches: 13729
	File system inputs: 8784
	File system outputs: 116672

	Command being timed: "tox -c coverage_tox.ini -e pytest-cov"
	User time (seconds): 43.38
	System time (seconds): 3.12
	Percent of CPU this job got: 77%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:00.20
	Maximum resident set size (kbytes): 104572
	Major (requiring I/O) page faults: 1
	Minor (reclaiming a frame) page faults: 447674
	Voluntary context switches: 1259
	Involuntary context switches: 15170
	File system inputs: 256
	File system outputs: 114776

	Command being timed: "tox -c coverage_tox.ini -e coverage-pycov"
	User time (seconds): 43.85
	System time (seconds): 3.10
	Percent of CPU this job got: 94%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:49.77
	Maximum resident set size (kbytes): 102292
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 1
	Minor (reclaiming a frame) page faults: 448188
	Voluntary context switches: 1080
	Involuntary context switches: 13637
	File system inputs: 8
	File system outputs: 114584

	Command being timed: "tox -c coverage_tox.ini -e coverage"
	User time (seconds): 19.54
	System time (seconds): 1.05
	Percent of CPU this job got: 91%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:22.41
	Maximum resident set size (kbytes): 102108
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 136861
	Voluntary context switches: 445
	Involuntary context switches: 7139
	File system inputs: 32
	File system outputs: 536

	Command being timed: "tox -c coverage_tox.ini -e pytest-cov"
	User time (seconds): 19.82
	System time (seconds): 1.10
	Percent of CPU this job got: 94%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:22.18
	Maximum resident set size (kbytes): 104724
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 132868
	Voluntary context switches: 426
	Involuntary context switches: 5302
	File system inputs: 0
	File system outputs: 536

	Command being timed: "tox -c coverage_tox.ini -e coverage-pycov"
	User time (seconds): 20.05
	System time (seconds): 1.05
	Percent of CPU this job got: 90%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:23.39
	Maximum resident set size (kbytes): 102204
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 139450
	Voluntary context switches: 446
	Involuntary context switches: 5167
	File system inputs: 0
	File system outputs: 536
```

**Comment by henri-hulski** [comment]
@faassen So I think its just a question of taste if we use coverage or pytest-cov.
I still prefer coverage, because it's more declarative, but if you want pytest-cov eg. because it's shorter, we can use that.

**Comment by faassen** [comment]
I'm leaning towards if it isn't broke then don't fix it in this case; since pytest-cov has worked until now we might as well keep it.

**Pull request merged**